src/sortStudents.ts

```diff
@@ -1,16 +1,79 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  type Sum = (a: number, b: number) => number;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  const sortedStudents: Student[] = [...students];
+  const sum: Sum = (a, b) => a + b;
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      if (order === 'asc') {
+        sortedStudents.sort((
+          firstStudent: Student,
```

Comment:
You don't need to specify type Student here, cause sortedStudents is already `Student[]`
```suggestion
          firstStudent,
```